### DISCUSSION TITLE
milvus 集群使用k8s如何水平拓展 bookies

### DISCUSSION THREAD

**Discussion by Gy1900** [milvus-io/milvus#34964] [Q&A and General discussion]
需求：
我之前milvus集群崩溃过，报错 bookie memory不够了,所以想要拓展 bookie，报错信息如下：“Not enough bookies to create ledger with ensembleSize=2, writeQuorumSize=2 and ackQuorumSize=2”
背景：
我用三主三从的6台服务器（每台机器8C16G） 以及k8s构建26个node来部署milvus 集群
目标：
1. 现在需要水平拓展，使其能够支持更多的数据，请问怎么操作？在这个里面https://milvus.io/docs/scaleout.md 我只看到了 heml的方式，并没有看到k8s的操作。
2. 还有个问题，我的存储结构是这样，但是 实际能存储的entity远远小于预想的数量，服务就会崩溃，请问是因为我的text也存下来了吗？这方面有没有什么建议？
![attu](https://github.com/user-attachments/assets/7ef5caa2-3b11-48b3-8e1d-a6b91d4b3322)
![collection](https://github.com/user-attachments/assets/a04c432a-2347-4779-9673-bf7937b74e7e)
![collection-col](https://github.com/user-attachments/assets/2c73bb5f-5af3-4ca5-8ef7-3c458cc69984)
![pods](https://github.com/user-attachments/assets/b8573083-2c7e-4477-bff8-342abc4d8402)

**Comment by locustbaby** [comment]
1. 你是用什么方式安装的整个集群呢？如果你是用helm安装的，那就用helm的方式扩容即可： https://milvus.io/docs/scale-dependencies.md#Increase-resources-per-Pulsar-bookie-pod
2. 预期存储多少量？向量和标量都会占用容量的，你是怎么预估的容量
3. 你说的6台服务器，和26个node 是什么意思？一共32台机器？

**Reply by Gy1900** [reply]
1. 没有用helm，就直接k8s + nfs
2. 预计两万份文档，大概82万行，我是根据已有的数据估计的，然后按照tool，我需要准备 171G的内存？
![image](https://github.com/user-attachments/assets/8ce9cffe-48eb-4bac-9bab-d657b2c654b9)
3. 使用K8s部署，6台机器做了3主3从，然后使用K8s 部署了proxy，etcd等26个节点的集群，实际一共6台机器，实际的48Core96G内存。
4. 我想问下，我需要在milvus的collection中存储text和vector，根据我的报错信息如下：“Not enough bookies to create ledger with ensembleSize=2, writeQuorumSize=2 and ackQuorumSize=2”，我是不是只需要 增加bookie节点就可以

**Reply by xiaofan-luan** [reply]
这么小的数据量建议直接使用standalone，没有必要走分布式。
总共只需要8GB内存就够了。

**Reply by xiaofan-luan** [reply]
使用docker compose安装即可，我感觉82w数据也不需要做任何横向扩展

**Comment by xiaofan-luan** [comment]
https://github.com/zilliztech/milvus-operator/blob/main/docs/administration/manage-dependencies/message-storage.md 
这个文档应该有具体关于bookeeper扩容的部分
